network/p2p/libp2pNode_test.go

```diff
@@ -259,13 +259,32 @@ func (suite *LibP2PNodeTestSuite) TestNoBackoffWhenCreatingStream() {
 
 	maxTimeToWait := maxConnectAttempt * maxConnectAttemptSleepDuration * time.Millisecond
 
-	var err error
-	unittest.RequireReturnsBefore(suite.T(), func() {
-		_, err = node1.CreateStream(context.Background(), *id2)
-	}, maxTimeToWait, fmt.Sprintf("create stream did not error within %d ms", maxTimeToWait))
+	// need to add some buffer time so that RequireReturnsBefore waits slightly longer than maxTimeToWait to avoid
+	// a race condition
+	someGraceTime := 100 * time.Millisecond
+	totalWaitTime := maxTimeToWait + someGraceTime
+
+	// make two separate stream creation attempt and assert that no connection back off happened
+	for i := 0; i < 2; i++ {
+		// Each CreateStream call makes maxConnectAttempt number of connection attempts within
+		// the context timeout duration - maxTimeToWait.
+		// However, the first connection attempt itself, in the create stream call may block for all of the maxTimeToWait
+		// on certain machine based on their setup never allowing a second connection attempt.
+		// Hence, the test makes the stream creation twice to ensure that a second connection attempt to the same node
+		// never fails with a backoff error
```

Comment:
In other setups the connection attempt fails immediately?

Reply:
@smnzhu On some machines (e.g. mine) the first connection times out (maybe due to firewall rules or other processes running) so we need a 2nd connection attempt to test that backoff isn't working